src/components/Layout/FormPageLayout.tsx

```diff
@@ -38,6 +42,19 @@ function FormPageLayoutComponent<F>(props: FormPageLayoutProps<F>) {
 
   const tids = useTestIds(props, "formPageLayout");
 
+  // Create a ref for each section here so we can coordinate both the `scrollIntoView`
+  // and the `IntersectionObserver` behaviors between the `LeftNav` and `FormSections` children
+  const sectionsWithRefs = useMemo(
+    () =>
+      formSections.map((section, id) => ({
+        section,
+        ref: createRef<HTMLElement>(),
+        // Unique key for each section to use in the observer
+        sectionKey: `section-${section.title ?? id}`,
+      })),
+    [formSections],
```

Comment:
Per prior conversations, it seems unlikely this `formSections` is ever going to be stable, but :shrug: also unclear how many re-renders the top-level FormPageLayout is really going to have anyway :thinking:

Reply:
yeah I guess we'll see, don't really see why it would need to change 🤔 even some of the "poll interval" type components that are in the app header won't be parent components to this so I'm guessing not an issue